### PR TITLE
Fix-Spectator Build and Break Block

### PR DESCRIPTION
**Pull request by SenseiTarzan**

### CODE REVIEW

File: src/main/java/cn/nukkit/Player.java [hunk 1]
```diff
@@ -2016,7 +2016,7 @@ protected void processLogin() {
 
         this.adventureSettings = new AdventureSettings(this)
                 .set(Type.WORLD_IMMUTABLE, isAdventure() || isSpectator())
-                .set(Type.WORLD_BUILDER, !isAdventure() && !isSpectator())
+                .set(Type.WORLD_BUILDER, !isAdventure() || !isSpectator())
```
Comment (joserobjr) [inline src/main/java/cn/nukkit/Player.java:2019]: That would be always true

GameMode | `not isAdventure` | `not isSpectator` | `(not isAdventure) OR (not isSpectator)`
---------|----|-----|--------
survival | true | true | true
creative | true | true | true
adventure | false | true | true
spectator | true | false | true

**Comment by lt-name** [comment]
https://github.com/CloudburstMC/Nukkit/pull/1572
Maybe it can be temporarily set to visitor permissions

File: src/main/java/cn/nukkit/Player.java (hunk 1 above)
Comment (SenseiTarzan) [reply src/main/java/cn/nukkit/Player.java:2019]: I don't know why but on my server it solved the spectator problem in spc and adventure gamemode and I could still break the blocks in creative and survival.

Comment (SenseiTarzan) [reply src/main/java/cn/nukkit/Player.java:2019]: GameMode | not isAdventure | not isSpectator | (not isAdventure) OR (not isSpectator)
-- | -- | -- | --
survival | true | true | true
creative | true | true | true
adventure | false | true | false
spectator | true | false | false

you are wrong 🤔🤔

**Comment by joserobjr** [comment]
Adding more context: 
This flag apparently can be toggle by /wb command when the education mode is enabled in vanilla.
https://minecraft.gamepedia.com/Commands/worldbuilder

"Toggles the ability to place and break blocks to enable or disable alternately."

It also affects Allow Deny and Border blocks:

https://minecraft.gamepedia.com/Allow_and_Deny
"When placed, allow blocks let users without world builder status place or destroy blocks in the area above them, while deny blocks prohibit users from placing or destroying blocks in the area above them."

https://minecraft.gamepedia.com/Border
 They prevent players who lack the worldbuilder ability from doing any of the following:

Jumping, climbing, tunneling, or swimming over or under them,
* Walking through them,
* Building (placing or destroying blocks) over or under them,
* Destroying the border block.

Adventure mode is also affected by this flag:
Players with the ability worldbuilder set to true can break and place blocks freely on Adventure Mode.‌[Bedrock and Education editions only]

The immutable world is also related to Education edition:
https://minecraft.gamepedia.com/Education_Edition
A new immutableWorld gamerule, if true it prevents any players who are not teachers (or who don't have worldbuilder permissions) from breaking or placing any blocks unless they are on top of an allow block.
Worldbuilder is an ability that can be applied to a player to allow bypassing the functions of border blocks and deny blocks.

**Comment by joserobjr** [comment]
The worldbuilder flag is automatically assumed if the player is OP in bedrock edition:

https://minecraft.gamepedia.com/Border
"In Education Edition, placing or breaking a border block requires the worldbuilder ability, which can be granted to the player by an operator or host. The equivalent in Bedrock Edition is operator permissions."

**Pull request closed without merging**